zlib-rs/src/inflate/bitreader.rs

```diff
@@ -120,7 +120,7 @@ impl<'a> BitReader<'a> {
         // SAFETY: assertion above ensures we have 8 bytes to read for a u64.
         let read = unsafe { core::ptr::read_unaligned(self.ptr.cast::<u64>()) }.to_le();
         self.bit_buffer |= read << self.bits_used;
-        let increment = (63 - self.bits_used) >> 3;
+        let increment = (63 ^ self.bits_used) >> 3;
```

Comment:
for future reference, this is correct assuming that `self.bits_used` is at most 63. We'll need to make sure that this is actually the case (I think it should be, and running this function is wasteful when the `bit_buffer` is already full. Just noting that; it's late over here right now so that needs a careful look later).

https://alive2.llvm.org/ce/z/boN8R8

Reply:
allright, I've looked at this: the only increment to `bits_used` is the bitwise or below, which would never exceed 63